velox/docs/develop/operators.rst

```diff
@@ -232,13 +232,77 @@ followed by the group ID column. The type of group ID column is BIGINT.
      - Description
    * - groupingSets
      - List of grouping key sets. Keys within each set must be unique, but keys can repeat across the sets.
+     - Grouping keys are specified with their output names.
    * - groupingKeyInfos
      - The names and order of the grouping key columns in the output.
    * - aggregationInputs
      - Input columns to duplicate.
    * - groupIdName
      - The name for the group-id column that identifies the grouping set. Zero-based integer corresponding to the position of the grouping set in the 'groupingSets' list.
 
+To illustrate why GroupingSets should use output column names lets examine the following SQL query:
+
+.. code-block:: sql
+
+  select COUNT(orderkey), count(distinct orderkey) from orders;
+
+In this query the user wants to compute aggregates on the same key, though with
+and without the DISTINCT clause. With a particular optimization strategy
+optimize.mixed-distinct-aggregations, Presto uses GroupIdNode to compute these.
+
+First, the optimizer duplicates every row assigning one copy to group 0 and another
+to group 1. This is achieved using the GroupIdNode with 2 grouping sets
+each using orderkey as a grouping key. In order to disambiguate the
+groups the orderkey column is aliased as a grouping key for one of the
+grouping sets.
+
+Lets say the orders table has 10 rows as follows:
+
+.. code-block::
```

Comment:
Would you share a screenshot of the docs? Curious how they look? Would it be better to format this as a table?

Reply:
![image](https://github.com/facebookincubator/velox/assets/87670623/c7389eee-c6e1-436f-bbe6-3a41cd5078ce)

Reply:
This seemed okay to me.